### ISSUE TITLE
PHP Warning – yii\base\ErrorException Undefined property: stdClass::$paging

### ISSUE THREAD

**Issue by samuel-clara** [scaramangagency/craftagram#76]
Using craftagram 4.0.0 with the basic example provided in the documentation:

```
{% set craftagram = craft.craftagram.getInstagramFeed() %}

{% if craftagram|length %}
    {% for item in craftagram.data %}
        <img src={{item.media_url}} />
    {% endfor %}
{% endif %}
```

I get:
PHP Warning – [yii\base\ErrorException](https://www.yiiframework.com/doc-2.0/yii-base-errorexception.html)
Undefined property: stdClass::$paging
 in /var/www/html/craft/vendor/jsmrtn/craftagram/src/services/CraftagramService.php
return parent::handleError($code, $message, $file, $line);

**Comment by jsmrtn** [comment]
Does the example feed show up in plugin settings? You should get feed preview under the `Save` button.

**Comment by samuel-clara** [comment]
In the plugin settings I get the same error. This does not happen on the production server, I guess because of the domain authorized. Is there a way to test the feed also on a local environment?

**Comment by jsmrtn** [comment]
Ah I see, this is on your local environment.

How are you testing on local environment at the moment? Are you seeing this error just from upgrading, or have you copied down the long access token from your production environment down to your local environment?

**Comment by samuel-clara** [comment]
I started a project from scratch using 4.0.0 and yes I've copied down the long access token from production to local environment.

**Comment by jsmrtn** [comment]
Interesting – I wouldn't have thought that the API would be overly concerned with the origin of the request providing it had valid tokens sent in the headers.

If you upgrade to `4.0.1` locally can you see if you get any extra logs outputted in `storage/logs/craftagram` that might point to where the error spawns from?

**Comment by samuel-clara** [comment]
Upgraded to 4.0.1 and revisit craftagram settings, same error but I can't see no craftagram logs under storage/logs.
The web.log file has outputted this:

2024-11-18 18:07:11 [web.INFO] [jsmrtn\craftagram\Craftagram::init] Craftagram plugin loaded {"memory":3280648} 
2024-11-18 18:07:11 [web.ERROR] [yii\base\ErrorException:2] yii\base\ErrorException: Undefined property: stdClass::$paging in /var/www/html/craft/vendor/jsmrtn/craftagram/src/services/CraftagramService.php:367
Stack trace:
#0 /var/www/html/craft/vendor/craftcms/cms/src/web/ErrorHandler.php(79): yii\base\ErrorHandler->handleError()
#1 /var/www/html/craft/vendor/jsmrtn/craftagram/src/services/CraftagramService.php(367): craft\web\ErrorHandler->handleError()
#2 /var/www/html/craft/vendor/jsmrtn/craftagram/src/variables/CraftagramVariable.php(33): jsmrtn\craftagram\services\CraftagramService->getInstagramFeed()
#3 /var/www/html/craft/vendor/twig/twig/src/Extension/CoreExtension.php(1802): jsmrtn\craftagram\variables\CraftagramVariable->getInstagramFeed()
#4 /var/www/html/craft/vendor/craftcms/cms/src/helpers/Template.php(148): Twig\Extension\CoreExtension::getAttribute()
#5 /var/www/html/craft/storage/runtime/compiled_templates/a9/a94f4b2beba34419883c836bb22307f5.php(226): craft\helpers\Template::attribute()
#6 /var/www/html/craft/vendor/twig/twig/src/Extension/CoreExtension.php(1967): __TwigTemplate_8f5e9bb7d671b2122cb0bdea5fbf10a7->{closure}()
#7 /var/www/html/craft/storage/runtime/compiled_templates/a9/a94f4b2beba34419883c836bb22307f5.php(70): Twig\Extension\CoreExtension::captureOutput()
#8 /var/www/html/craft/vendor/twig/twig/src/Template.php(393): __TwigTemplate_8f5e9bb7d671b2122cb0bdea5fbf10a7->doDisplay()
#9 /var/www/html/craft/vendor/twig/twig/src/Template.php(349): Twig\Template->yield()
#10 /var/www/html/craft/vendor/twig/twig/src/Template.php(364): Twig\Template->display()
#11 /var/www/html/craft/vendor/twig/twig/src/TemplateWrapper.php(35): Twig\Template->render()
#12 /var/www/html/craft/vendor/twig/twig/src/Environment.php(306): Twig\TemplateWrapper->render()
#13 /var/www/html/craft/vendor/craftcms/cms/src/web/View.php(539): Twig\Environment->render()
#14 /var/www/html/craft/vendor/craftcms/cms/src/web/View.php(592): craft\web\View->renderTemplate()
#15 /var/www/html/craft/vendor/craftcms/cms/src/web/TemplateResponseFormatter.php(57): craft\web\View->renderPageTemplate()
#16 /var/www/html/craft/vendor/yiisoft/yii2/web/Response.php(1109): craft\web\TemplateResponseFormatter->format()
#17 /var/www/html/craft/vendor/craftcms/cms/src/web/Response.php(339): yii\web\Response->prepare()
#18 /var/www/html/craft/vendor/yiisoft/yii2/web/Response.php(340): craft\web\Response->prepare()
#19 /var/www/html/craft/vendor/yiisoft/yii2/base/Application.php(390): yii\web\Response->send()
#20 /var/www/html/public_html/index.php(12): yii\base\Application->run()
#21 {main} {"memory":6108568,"exception":"[object] (yii\\base\\ErrorException(code: 2): Undefined property: stdClass::$paging at /var/www/html/craft/vendor/jsmrtn/craftagram/src/services/CraftagramService.php:367)"} 
2024-11-18 18:07:11 [web.INFO] [application] Request context:

**Comment by jsmrtn** [comment]
It only outputs errors into `craftagram.log` if you get failures, so I guess you're getting nothing out of `mediaIDs` for some reason, but not in a way that causes the `try {} catch {}` to fail.

Presuming this is still a localhost issue, can you go to `vendor/jsmrtn/craftagram/src/services/CraftagramService.php` and add `var_dump($mediaIDs);exit;` on line 333, then refresh the settings page and paste in the response. I'm assuming it's erroring in a way I'm not handling, because I'm not seeing this error – `paging` does exist under `mediaIDs`..

```
// $mediaIDs->paging

object(stdClass)#726 (2) { ["cursors"]=> object(stdClass)#727 (2) {
 ["before"]=> string(126) "QVFIUnZAHczJPMGpST043cVF6bVJQVlU4ZADNLMEllVTR4UGJMeDYtZAmhnQTZA1cDdDQ1pZAMmhxUUU0QW9TWjl4ZAmY0aHVNVnp1MUoyRkNPM0NyTXdhb05pOVVB"
 ["after"]=> string(124) "QVFIUmJJUXgxMllzVTQyc2c5TDZA3MGpXR0lMSEhJZAUpWMFVIZAHYzZAm1qR0NKU3pXaVVCaXJGSHhUa2VvbHR6a05DSGtxUmVrRHdNMkF1QkFnc05fTlhLcTBn" 
} 
```

**Comment by samuel-clara** [comment]
Ok this is very helpful thank you, I will investigate on the local environment then.

I confirm the error happens only locally.

dump response:
object(stdClass)#1724 (1) { ["data"]=> array(0) { } }

**Comment by jsmrtn** [comment]
Well that's... something. I mean, it's right not to throw errors, because it's _not_ erroring – you're just not getting any media IDs returned from the [media endpoint](https://developers.facebook.com/docs/instagram-platform/instagram-graph-api/reference/ig-user/media/#reading).

What I can't figure out is why this doesn't work for you _locally_ but does work for you on the promoted environment. I don't think this is an issue with instagram caring about the origin URL of the API call, as when testing I use an ngrok URL from `dev share` to allow my localhost environment to sync up with instagram to get the callback post. But if I visit the `.ddev.site` url then it carries on working fine.

I guess the easiest option might be to just start throwing `var_dump` after each cURL request in `CraftagramService` and seeing what the responses say, it might lead you (and me) to the direction.

**Comment by samuel-clara** [comment]
Ok thank you, I will try to understand what is the cause on my side. But maybe the plugin should not respond with a php warning neither on the front end nor on the control panel when there are any media IDs!?

**Comment by jsmrtn** [comment]
Agreed, it should fail silently and log an error. Fixed in 4.0.2 👍
